content/scripts/build.js

```diff
@@ -1753,6 +1723,45 @@ class Builder {
 
     // With the sidebar out of the way, go ahead and check the rest
     this.injectFlaws(source, doc, $, rawContent);
+    if (this.options.fixFlaws) {
```

Comment:
It seems that for the case when `source.htmlAlreadyRendered` is `true`, you will reach this point with `newRawHtml` being `null` and you'd better skip the fixing of flaws or else `replaceMatchesInText()` will throw an error due to `newRawHtml` being `null`? So perhaps best to use:
```suggestion
    if (newRawHtml && this.options.fixFlaws) {
```

Reply:
You're absolutely right. I was able to produce the bug with
```
node content build -l en-us --fix-flaws --fix-flaws-dry-run -f gecko --no-progressbar
```
and I have, in my `.env`:
```
▶ cat .env | rg ARCHIVE
BUILD_ARCHIVE_ROOT=archivecontent/files
```

Fixed now.